File: src/DynamicLinkLibraries/DataPerformer/DataPerformer.Portable/DifferentialEquationProcessors/EulerProcessor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DataPerformer.Interfaces;
using DataPerformer.Portable.Measurements;

namespace DataPerformer.Portable.DifferentialEquationProcessors
{
    /// <summary>
    /// Euler processor
    /// </summary>
    public class EulerProcessor : DifferentialEquationProcessor
    {
        /// <summary>
        /// Singleton
        /// </summary>
        new static public readonly DifferentialEquationProcessor Processor = new EulerProcessor();

        /// <summary>
        /// Auxiliary variable
        /// </summary>
        private double[] w;

        /// <summary>
        /// Constructor
        /// </summary>
        private EulerProcessor()
        {
        }



        /// <summary>
        /// Performs step of integration
        /// </summary>
        /// <param name="t0">Step start</param>
        /// <param name="t1">Step finish</param>
        public override void Step(double t0, double t1)
        {
            isBusy = true;
            if (Dim == 0)
            {
                return;
            }
            double dt = t1 - t0;
            int i = 0;
            double t = t0;
            foreach (IMeasurements m in equations)
            {
                for (int j = 0; j < m.Count; j++)
                {
                    w[i] = (double)m[j].Parameter();
                    ++i;
                }
            }
            StaticExtensionDataPerformerPortable.Time = t;
            StaticExtensionDataPerformerPortable.Desktop.ResetUpdatedMeasurements();
            UpdateMeasurements();
            i = 0;
            foreach (IMeasurements m in equations)
            {
                IDifferentialEquationSolver s = m as IDifferentialEquationSolver;
                s.CalculateDerivations();
                for (int j = 0; j < m.Count; j++)
                {
                    IDerivation der = m[j] as IDerivation;
                    w[i] = w[i] + der.Derivation.ToDouble() * dt;
                    ++i;
                }
                s.CopyVariablesToSolver(i - m.Count, w);
            }
            isBusy = false;
        }

        /// <summary>
        /// Sets consumers
        /// </summary>
        /// <param name="collection">Consumers</param>
        /// <returns>Lists of parameters</returns>
        public override void Set(object collection)
        {
            base.Set(collection);
            UpdateDimension();
        }

        /// <summary>
        /// Updates dimension
        /// </summary>
        public override void UpdateDimension()
        {
            int n = Dim;
            w = new double[n];
        }

        /// <summary>
        /// Creates new processor
        /// </summary>
        public override IDifferentialEquationProcessor New
        {
            get
            {
                return new EulerProcessor();
            }
        }

    }
}

File: src/TypeScript/TypeScriptLibrary/Library/Measurements/DifferentialEquations/Processors/EulerProcessor.ts
import { IMeasurements } from "../../Interfaces/IMeasurements";
import { IDifferentialEquationProcessor } from "../Interfaces/IDifferentialEquationProcessor ";
import { DifferentialEquationProcessor } from "./DifferentialEquationProcessor";

export class EulerProcessor extends DifferentialEquationProcessor
{

    w: number[] = [];

 

    stepDifferentialEquations(start: number, finish: number): void
    {
        let dt = finish - start;
        let i = 0;
        for (let m of this.measurements)
        {
            m.updateMeasurements();
            for (let j = 0; j < m.getMeasurementsCount(); j++)
            {
                var mea = m.getMeasurement(j);
                var x = mea.getMeasurementValue();
                this.w[i] = this.performer.convertFromAny<number>(x);
                ++i;
            }
        }
        i = 0;
        for (let s of this.equations)
        {
            s.calculateDerivations();
            let m = s as unknown as IMeasurements;
            let count = m.getMeasurementsCount();
            for (var j = 0; j < count; j++)
            {
                var mea = m.getMeasurement(j);
                var v = this.performer.getDerivationMeasurement(mea);
                var y = this.performer.convertFromAny<number>(v);
                this.w[i] +=  y * dt;
                ++i;
            }
            s.copyVariablesToSolver(i - count, this.w);
        }

    }

    updateDimension(): void
    {
        super.updateDimension();
        this.w = new Array(this.dimension);
    }

    newDifferentialEquations(): IDifferentialEquationProcessor
    {
        return new EulerProcessor();
    }

}